coral/src/app/components/ComplexNativeSelect.tsx

```diff
@@ -0,0 +1,73 @@
+import { NativeSelect } from "@aivenio/aquarium";
+import { ChangeEvent } from "react";
+
+type ComplexNativeSelectProps<T> = {
+  options: Array<T>;
+  onBlur: (option: T | undefined) => void;
+  optionToString: (opt: T) => string;
+  getValue: (opt: T) => string;
+  placeholder: string;
+  labelText: string;
+  disabled?: boolean;
+  defaultValue?: string;
+  required?: boolean;
+  error?: string;
+};
+
+function ComplexNativeSelect<T>(props: ComplexNativeSelectProps<T>) {
+  const {
+    options,
+    onBlur,
+    disabled = false,
+    required = false,
+    defaultValue,
+    labelText,
+    optionToString,
+    placeholder,
+    getValue,
+    error,
+  } = props;
+
+  // the placeholder behavior will be covered by DS NativeSelect
+  // soon, this is a temp solution
+  const placeholderValue = "055d87f2-8cd4-11ed-a1eb-0242ac120002";
+  function setNewOption({ target: { value } }: ChangeEvent<HTMLSelectElement>) {
+    if (value === placeholderValue) {
+      onBlur(undefined);
+    } else {
+      const newOption = options.find((option) => getValue(option) === value);
+      if (newOption) {
+        onBlur(newOption);
+      }
+    }
+  }
+
+  return (
+    <NativeSelect
+      onBlur={setNewOption}
+      disabled={disabled}
+      labelText={labelText}
+      required={required}
+      defaultValue={defaultValue || placeholderValue}
```

Comment:
The need for `defaultValue` is not clear to me? Should it actually be simply `value`.

Reply:
DS `NativeSelect` will take care of placeholder value soon, so there's no need for us to add a custom handling for that in a different way. That's why I didn't want to tie handling of the placeholder text into our functionality too deep.